### DISCUSSION TITLE
Installing prerequisites

### DISCUSSION THREAD

**Discussion by ryani42** [neams-th-coe/cardinal#565] [Q&A]
Hi, I previously have been using OpenMC, which I have installed as a conda environment. Im attempting to install moose which requires mpich. 

Ive installed mpich following the instructions here https://www.mpich.org/static/downloads/4.1/mpich-4.1-README.txt. However when I enter the command 'mpicc --version', the command line outputs : '/home/ryan/anaconda3/bin/mpicc: line 285: x86_64-conda_cos6-linux-gnu-cc: command not found'. Im not sure how to resolve this problem as im unsure whether mpicc has installed properly.

The files all look in order however I dont want to move on with the next stages of installation without being sure. 

Im fairly new to linux in general so any assistance would be greatly appreciated.

**Comment by aprilnovak** [comment]
Hi @ryani42, welcome! Are you trying to build Cardinal with MOOSE's conda environment, or "from scratch"? If it's the former, then if you follow the instructions to get [MOOSE's conda environment](https://mooseframework.inl.gov/getting_started/installation/conda.html), all of the dependencies (MPI, HDF5, etc.) should be taken care of automatically. I'd recommend that route if it works for you.

**Reply by aprilnovak** [reply]
Addressing a separate part of your question, the `which` command will only be able to find `mpicc` if it's on your `PATH`. So it's possible that you did indeed install mpich fine, but you just need to add that location to your `PATH`, like `export PATH=/path/to/mpich`.

**Reply by ryani42** [reply]
Hi, im attempting to construct it from scratch due to the incompatibility with NekRS.  Apologies im new to linux. My mpich installation is in the folder called mpich-install.  The mpich documentation says the following:
![image](https://user-images.githubusercontent.com/22966493/216120627-bf0a2676-58e8-45ec-ba28-2f8df1d42bb1.png)
Once I enter 'which mpicc' and 'which mpiexec' in accordance with the documentation and that all seems to work fine. However when I enter the commands on the prerequisite page:
![image](https://user-images.githubusercontent.com/22966493/216122915-6f15b31c-aef7-4144-b8a0-3e5629fc7451.png)
I do not get the response from the console which I should. Should the installation of mpich mean that I get the results shown in the screenshot above? Because if so that is not happening.

I also saw that I can add the export line to the bashrc file, however I have not found any useful information about how I can do this. Could you point me to a helpful resource?

**Reply by aprilnovak** [reply]
Yes, you'll need to do the non-conda approach if you want to use NekRS - so you're on the right track. I only saw you mention OpenMC in your first post, so I thought we might have been able to get away with the conda route :) 

Your bashrc is a file that is executed every time you open a shell. You'll need to edit that file. If you use vim text editor:

```
vi ~/.bashrc

# this opens the file. Add this line at the end of the file

export PATH=/path/to/mpi:$PATH
```

and then save and close the file. Now, every time you open a new terminal window, your PATH will have that extra location added to it.

Can you show me what you see with this:

```
ls /path/to/mpi/bin
```

**Reply by ryani42** [reply]
Ive added this line to the bash,
![image](https://user-images.githubusercontent.com/22966493/216134779-7dabf0f5-5c3e-40fa-8a6d-71c3b9ab49ec.png)
I added 3 because I just wanted to cover all bases as I wasnt 100 percent sure :).
This is the output:
![image](https://user-images.githubusercontent.com/22966493/216135353-c4715d65-b89b-4384-a606-11b1d417ee50.png)

And im also now aware that the location is just /mpich-install !

Thank you so much for all the help!

**Reply by aprilnovak** [reply]
Yes, `/path/to/mpi` is not a real location, it's just my way of trying to explain to "use whatever path to MPI you have on your machine." 

If something is missing in your dependencies, when you try to build Cardinal it will fail - so I think you can continue through the instructions and post any further questions you have here.

**Reply by ryani42** [reply]
Ive moved on with the installation and when I try to run the line './contrib/moose/scripts/update_and_rebuild_petsc.sh' I get the following error:

![image](https://user-images.githubusercontent.com/22966493/216327203-371f91b8-b066-4b43-961e-f3e2ae9eadc4.png)

Ive tried to look up a solution but have found no help. I assume this is a problem with my mpi installation however I have followed everything exactly as stated in the mpich installation so im unsure what could be causing this.

**Reply by ryani42** [reply]
Hi, I have now solved the above problem but have of course run into a further problem. When running the same command as above I now get this error:
![image](https://user-images.githubusercontent.com/22966493/216336776-4f122dc2-54cd-463f-a64e-0e5293048568.png)


EDIT: I have created a new thread so the title is more relevant.
